### PR TITLE
Convert windows CI to stages

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Convert windows ci yaml to use stages which allows better reporting and dependency management.

If the new UI is enable in Azure DevOps, this change will look like this:

image

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: .vsts-ci/windows-daily.yml [hunk 1]
```diff
@@ -6,23 +6,29 @@ trigger:
     include:
     - master
     - release*
+    - feature*
   paths:
     include:
     - '*'
     exclude:
+    - /tools/releaseBuild/**/*
```
Comment (adityapatwardhan) [inline .vsts-ci/windows-daily.yml]: Fix paths

File: .vsts-ci/windows-daily.yml [hunk 2]
```diff
@@ -6,23 +6,29 @@ trigger:
...
   branches:
     include:
     - master
     - release*
+    - feature*
   paths:
     include:
     - '*'
     exclude:
+    - /tools/releaseBuild/**/*
```
Comment (adityapatwardhan) [inline .vsts-ci/windows-daily.yml]: Same as above

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 4 hours

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 4 hours ago

**Pull request merged**